packages/sanity/src/structure/diffView/hooks/useDiffViewRouter.ts

```diff
@@ -0,0 +1,90 @@
+import {fromPairs, toPairs} from 'lodash'
+import {useCallback} from 'react'
+import {useRouter} from 'sanity/router'
+
+import {
+  DIFF_SEARCH_PARAM_DELIMITER,
+  DIFF_VIEW_NEXT_DOCUMENT_SEARCH_PARAMETER,
+  DIFF_VIEW_PREVIOUS_DOCUMENT_SEARCH_PARAMETER,
+  DIFF_VIEW_SEARCH_PARAMETER,
+} from '../constants'
+import {type DiffViewMode} from '../types/diffViewMode'
+
+type NavigateDiffView = (
+  options: {
+    mode?: DiffViewMode
+  } & Partial<
```

Comment:
Would it be slightly more readable if options was typed `{mode?: DiffViewMode, previousDocument?: {type: string, id: string}}, nextDocument?: ?: {type: string, id: string}}`